src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.cs

```diff
@@ -28,47 +32,59 @@ internal static partial class ValueSetFactory
         internal static readonly IValueSetFactory<int> ForNint = NintValueSetFactory.Instance;
         internal static readonly IValueSetFactory<uint> ForNuint = NuintValueSetFactory.Instance;
 
-        public static IValueSetFactory? ForSpecialType(SpecialType specialType, bool isNative = false)
+        internal static readonly IValueSet PositiveIntValues = new NumericValueSet<int, IntTC>(0, int.MaxValue);
+
+        public static IValueSet Shift(IValueSet values, int offset)
```

Comment:
:memo: This can be possibly generalized to all numeric types. 
<details><summary>IValueSet Shift(T)</summary>

```cs
            public IValueSet Shift(T offset)
            {
                TTC tc = default;
                if (tc.Related(GreaterThan, offset, tc.Zero))
                {
                    var builder = ArrayBuilder<(T first, T last)>.GetInstance(_intervals.Length);
                    foreach(var interval in _intervals)
                    {
                        var (first, last) = interval;
                        while (tc.Related(LessThan, last, tc.MaxValue))
                        {
                            offset = tc.Prev(offset);
                            first = tc.Next(first);
                            last = tc.Next(last);
                            if (tc.Related(Equal, offset, tc.Zero))
                                break;
                        }
                        builder.Add((first, last));
                    }
                    return new NumericValueSet<T, TTC>(builder.ToImmutableAndFree());
                }

                if (tc.Related(LessThan, offset, tc.Zero))
                {
                    var builder = ArrayBuilder<(T first, T last)>.GetInstance(_intervals.Length);
                    foreach(var interval in _intervals)
                    {
                        var (first, last) = interval;
                        while (tc.Related(GreaterThan, first, tc.MinValue))
                        {
                            offset = tc.Next(offset);
                            first = tc.Prev(first);
                            last = tc.Prev(last);
                            if (tc.Related(Equal, offset, tc.Zero))
                                break;
                        }
                        builder.Add((first, last));
                    }
                    return new NumericValueSet<T, TTC>(builder.ToImmutableAndFree());
                }

                return this;
            }
```

</details>

The logic is inlined for int.